quesma/quesma/schema_transformer.go

```diff
@@ -526,6 +526,32 @@ func (s *SchemaCheckPass) applyTimestampField(indexSchema schema.Schema, query *
 
 }
 
+func (s *SchemaCheckPass) applyFieldEncoding(indexSchema schema.Schema, query *model.Query) (*model.Query, error) {
+
+	visitor := model.NewBaseVisitor()
+
+	var err error
+
+	visitor.OverrideVisitColumnRef = func(b *model.BaseExprVisitor, e model.ColumnRef) interface{} {
+		if _, ok := indexSchema.ResolveField(e.ColumnName); ok {
+			return model.NewColumnRef(util.FieldToColumnEncoder(e.ColumnName))
```

Comment:
same here we could use `indexSchema schema.Schema` struct

Reply:
Added comment